packages/babel-traverse/src/path/ancestry.js

```diff
@@ -33,7 +33,11 @@ export function find(callback): ?NodePath {
  */
 
 export function getFunctionParent(): ?NodePath {
-  return this.findParent(path => path.isFunction() || path.isProgram());
+  const parent = this.findParent(path => path.isFunction() || path.isProgram());
+  if (parent.isProgram()) {
```

Comment:
@jridgewell umm didn't follow.. 🤔 

You meant just to do something like: `return parent.isFunction() ? parent : null;`?

I added the `isProgram` check explicitly so I could throw an error...